cmem/cmem.go

```diff
@@ -0,0 +1,16 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package cmem provides helper functionality for accessing memory in a C
+// compatible fashion.
+package cmem
+
+// CMarshaler is an interface for types that can marshal their data into an
+// C compatible binary layout. This can be particularly interesting when used in
+// combination with code generation. Furthermore this is a more efficient way of
+// appending data, as it is more efficient than the reflection-based way of
+// doing so which is used under when unmarshalled data is used with Append().
```

Comment:
not sure the comment about `Append()` makes sense in this new general context.

Reply:
Removed the part about `Append()`. Makes it a bit easier to read now, I guess...